### DISCUSSION TITLE
data reading system for the sensor (Freestyle Libre 2)

### DISCUSSION THREAD

**Discussion by Lloansi** [NightscoutFoundation/xDrip#3553] [General]
Hello, my name is Lloansi, and I am a mobile app developer. Together with a group of friends, we are developing an app for glucose monitoring, and we came across this repository.

On our own, we are unable to access the system used by the Freestyle Libre 2, so it would be incredibly helpful if someone could assist us in creating a data reading system for the sensor. We would be very grateful to be able to launch this non-profit app for people with diabetes.

**Comment by Navid200** [comment]
xDrip is non for profit for diabetics already.
Contribute to xDrip instead of creating an alternative.
Or, explain your justification for why there is a need for a differentiator.

**Reply by Lloansi** [reply]
It's not for profit, as I mentioned; we are a group of researcher friends with an educational purpose, aiming to create an application to gather data from glucometers, which can then be used for making future predictions with trained models.

The reason for using a differentiator app is mainly because we need to conduct trials and send this information. This means:

We want to improve the UI/UX to make it easier for patients to use this app.
As mentioned, we need to implement functionalities for sending this information.
I am not proficient in Java, so I want to adapt the necessary parts of the code to Kotlin Jetpack Compose for receiving and sending information to/from OOP2.

If someone could explain how OOP2 works and how it interacts with the application, it would greatly help us.

Does anyone know if I can contact @tzachi-dar or @jamorham directly?

**Reply by Lloansi** [reply]
Otherwise, I also have the idea of contributing the data collection to the project.

That is, implementing a system in xDrip for sending data. What do you think, @Navid200 ?

That way, I could contribute to the official project.

**Reply by Navid200** [reply]
You are welcome to open PRs (submissions).
This is an open source project.  Everyone is welcome and encouraged to contribute.

**Reply by Lloansi** [reply]
Perfect, I will try to implement it in the coming days. Do you know where I can find where the information is stored? That is, where all the information sent by the sensor is saved so that it can be sent later.
